### PR TITLE
[DB] disable write stalling and stopping due to pending compaction bytes for index cfs

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Index db cfs can become large and require compactions. Write throttling based on compaction bytes can happen after initial ingestion of data, and is hard to investigate. Disable write throttling for index cfs.

## Test plan 

Deploy to private testnet.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/typed-store/src/rocks/mod.rs [hunk 1]
```diff
@@ -2428,7 +2428,14 @@ pub fn default_db_options() -> DBOptions {
         read_size_from_env(ENV_VAR_DB_WAL_SIZE).unwrap_or(DEFAULT_DB_WAL_SIZE) as u64 * 1024 * 1024,
     );
 
+    // Num threads for compaction and flush.
     opt.increase_parallelism(4);
+
+    // Disable write stalling and stopping based on pending compaction bytes.
+    // Throttling writes in Sui does not reduce the incoming data throughput most of the time,
+    // especially on the larger column families that do not get deleted per epoch.
+    opt.set_hard_pending_compaction_bytes_limit(0);
```
Comment (phoenix-o) [inline crates/typed-store/src/rocks/mod.rs:2437]: do you want to adjust soft_pending_compaction_bytes_limit as well?

Comment (mwtian) [reply crates/typed-store/src/rocks/mod.rs:2438]: Yes. Somehow that change was lost.

**Comment by halfprice** [comment]
Should we only apply this change to fullnode? I wonder if there is no back pressure from DB to validators, it may enter a state that is hard to recovery (e.g. OOM along with large pending compaction).

**Comment by mystenmark** [comment]
> Throttling writes based on compaction bytes will not help reduce write rate especially on full nodes.

I don't understand this, how does it not reduce write rate? On validators, once object table writes stop, overload protection will kick in, and signing will stop.

Overall it is very hard to understand what the effects of this change might be, and I echo Zhe's concern

**Comment by mwtian** [comment]
I agree throttling writes on the object table may help in some cases, but right now write throttling really lacks observability and happens unpredictably, much later after actual workload. So far in experiments flow control happens earlier than DB write throttling, which is more desirable. How about only turn off write throttling for index cfs?

**Comment by mystenmark** [comment]
> I agree throttling writes on the object table may help in some cases, but right now write throttling really lacks observability and happens unpredictably, much later after actual workload. So far in experiments flow control happens earlier than DB write throttling, which is more desirable. How about only turn off write throttling for index cfs?

That sounds good to me. I think the main downside would be increased read amplification, but as I understand it the very largest index cfs are rarely used anyway.

**Review by mystenmark — APPROVED** [review]

**Pull request merged**